### PR TITLE
Issue #3487312: Site managers can't see content of groups they are not part of

### PR DESCRIPTION
**Pull request by vcsvinicius**
## Problem (for internal)
The site-managers is not able to see the content in groups that they are not part of. So as for now, the group membership kind of outweighs the site manager status, which should not be the case.

## Solution (for internal)
We are using 'bypass group access' permission from group module and it was removed after version 2, check [#3259076](https://www.drupal.org/project/group/issues/3259076)

So I changed the validation to check group-permission from logged user.

## Release notes (to customers)
Fixed access from site-manager at group contents without be part of them.

## Issue tracker
[PROD-31123](https://getopensocial.atlassian.net/browse/PROD-31123)
[#3487312](https://www.drupal.org/project/social/issues/3487312)

## Theme issue tracker
N/A

## How to test
- [ ] Create a group
- [ ] Create some posts private to group and others for community
- [ ] Log-in with site-managers not-member of group created before
- [ ] Go to group and check the posts, the site-managers should be able to see all content from group

## Change Record
N/A

## Translations
N/A

[PROD-31123]: https://getopensocial.atlassian.net/browse/PROD-31123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4184-gxjbn9ss0ntdckb0r59fhga5eabyg9ju.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6734ed5a214e5b1c0507b27d

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4184-gxjbn9ss0ntdckb0r59fhga5eabyg9ju.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6734ed5a214e5b1c0507b27d

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4184-gxjbn9ss0ntdckb0r59fhga5eabyg9ju.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6734ed5a214e5b1c0507b27d

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4184-gxjbn9ss0ntdckb0r59fhga5eabyg9ju.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6734ed5a214e5b1c0507b27d

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4184-gxjbn9ss0ntdckb0r59fhga5eabyg9ju.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6734ed5a214e5b1c0507b27d

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4184-gxjbn9ss0ntdckb0r59fhga5eabyg9ju.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6734ed5a214e5b1c0507b27d

File: modules/custom/activity_viewer/src/Plugin/views/filter/ActivityPostVisibilityAccess.php [hunk 1]
```diff
@@ -205,7 +224,11 @@ public function query():void {
     $post_access = new Condition('AND');
     $post_access->condition('activity__field_activity_entity.field_activity_entity_target_type', 'post');
 
-    if (!$account->hasPermission('bypass group access')) {
+    // Get group from url-parameter.
+    $group = $this->routeMatch->getParameter('group');
```
Comment (denis-getopensocial) [inline modules/custom/activity_viewer/src/Plugin/views/filter/ActivityPostVisibilityAccess.php:228]: you are lucky that https://github.com/goalgorilla/open_social/pull/4118 is not merged yet, it brings a current group service that could have been used. :)

**Review by denis-getopensocial — APPROVED** [review]
👍

**Comment by denis-getopensocial** [comment]
Tested and all good.
Screenshot done as Site Manager not belonging to the group.
Screenshot 2024-12-03 at 14 46 24

**Pull request merged**